java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java

```diff
@@ -1290,7 +1290,12 @@ private void clearStreams() {
   private void readStripe() throws IOException {
     StripeInformation stripe = beginReadStripe();
     planner.parseStripe(stripe, fileIncluded);
-    includedRowGroups = pickRowGroups();
+    try {
+      includedRowGroups = pickRowGroups();
+    } catch (Throwable e) {
```

Comment:
this feels too broad to swallow any throwable. If the aim is to just deal with inconsistencies of not seeing an index stride and not seeing indexes.

I think a better change is inside the method `pickRowGroups` where after reading rowIndexes we test if row indexes are absent and treat that case as no row groups to skip.

Reply:
Thank you @pavibhai 
My question is when `pickRowGroups `fails (not only the inconsistencies issue) should we continue to read the file ? Or throw an exception directly?

Reply:
Let me give a split answer on this:

* When the inconsistent issue happens should we continue reading? Yes as this is well-defined on what the issue is i.e. row index stride is set but indexes have not been written.
* Should other issues in reading indexes be intercepted? The general answer here is no, as we don't know what the issue is that you are encountering so we don't know if it is correct to intercept it.

Reply:
OK, thank you @pavibhai.